odo/backends/tests/test_pandas.py

```diff
@@ -37,3 +39,11 @@ def test_datetime_to_timestamp():
     ts = odo(dt, pd.Timestamp)
     assert isinstance(ts, pd.Timestamp)
     assert ts == pd.Timestamp('2014-01-01')
+
+
+def test_nan_to_nat():
+    assert odo(float('nan'), pd.Timestamp) is pd.NaT
+
```

Comment:
kk, though np.nan is just `PyFloat_FromDouble((float) 0x7fc00000UL)` so it is not a custom object.